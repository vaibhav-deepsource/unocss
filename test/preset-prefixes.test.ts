import { createGenerator } from '@unocss/core'
import presetUno from '@unocss/preset-uno'
import presetIcons from '@unocss/preset-icons'
import presetAttributify from '@unocss/preset-attributify'
import { expect, test } from 'vitest'

const targets = [
  'icon-mdi-alarm',
  '[uno-sm~="bg-blue-400"]',
]

test('options properties does not override each other', async() => {
  const uno1 = createGenerator({
    presets: [
      presetAttributify({ prefix: 'uno-' }),
      presetUno(),
      presetIcons({ prefix: 'icon-' }),
    ],
  })

  const { css: css1 } = await uno1.generate(new Set(targets))
  expect(css1).toMatchSnapshot()

  const uno2 = createGenerator({
    presets: [
      presetIcons({ prefix: 'icon-' }),
      presetUno(),
      presetAttributify({ prefix: 'uno-' }),
    ],
  })

  const { css: css2 } = await uno2.generate(new Set(targets))
  expect(css2).toMatchSnapshot()
})
