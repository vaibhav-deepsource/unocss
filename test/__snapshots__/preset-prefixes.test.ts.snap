// Vitest Snapshot v1

exports[`options properties does not override each other 1`] = `
"/* layer: icons */
.icon-mdi-alarm{--un-icon:url(\\"data:image/svg+xml;utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' width='1em' height='1em' preserveAspectRatio='xMidYMid meet' viewBox='0 0 24 24'%3E%3Cpath d='M12 20a7 7 0 0 1-7-7a7 7 0 0 1 7-7a7 7 0 0 1 7 7a7 7 0 0 1-7 7m0-16a9 9 0 0 0-9 9a9 9 0 0 0 9 9a9 9 0 0 0 9-9a9 9 0 0 0-9-9m.5 4H11v6l4.75 2.85l.75-1.23l-4-2.37V8M7.88 3.39L6.6 1.86L2 5.71l1.29 1.53l4.59-3.85M22 5.72l-4.6-3.86l-1.29 1.53l4.6 3.86L22 5.72z' fill='currentColor'/%3E%3C/svg%3E\\");mask:var(--un-icon) no-repeat;mask-size:100% 100%;-webkit-mask:var(--un-icon) no-repeat;-webkit-mask-size:100% 100%;background-color:currentColor;height:1em;width:1em;}
/* layer: default */
@media (min-width: 640px){
[uno-sm~=\\"bg-blue-400\\"]{--un-bg-opacity:1;background-color:rgba(96,165,250,var(--un-bg-opacity));}
}"
`;

exports[`options properties does not override each other 2`] = `
"/* layer: icons */
.icon-mdi-alarm{--un-icon:url(\\"data:image/svg+xml;utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' width='1em' height='1em' preserveAspectRatio='xMidYMid meet' viewBox='0 0 24 24'%3E%3Cpath d='M12 20a7 7 0 0 1-7-7a7 7 0 0 1 7-7a7 7 0 0 1 7 7a7 7 0 0 1-7 7m0-16a9 9 0 0 0-9 9a9 9 0 0 0 9 9a9 9 0 0 0 9-9a9 9 0 0 0-9-9m.5 4H11v6l4.75 2.85l.75-1.23l-4-2.37V8M7.88 3.39L6.6 1.86L2 5.71l1.29 1.53l4.59-3.85M22 5.72l-4.6-3.86l-1.29 1.53l4.6 3.86L22 5.72z' fill='currentColor'/%3E%3C/svg%3E\\");mask:var(--un-icon) no-repeat;mask-size:100% 100%;-webkit-mask:var(--un-icon) no-repeat;-webkit-mask-size:100% 100%;background-color:currentColor;height:1em;width:1em;}
/* layer: default */
@media (min-width: 640px){
[uno-sm~=\\"bg-blue-400\\"]{--un-bg-opacity:1;background-color:rgba(96,165,250,var(--un-bg-opacity));}
}"
`;
